src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -204,749 +204,114 @@ private ImmutableArray<ISwitchCase> GetPatternSwitchStatementCases(BoundPatternS
-                        break;
-
-                    case BinaryOperatorKind.GreaterThanOrEqual:
-                        switch (operatorKind & BinaryOperatorKind.TypeMask)
-                        {
-                            case BinaryOperatorKind.Int:
-                            case BinaryOperatorKind.Long:
-                                return BinaryOperationKind.IntegerGreaterThanOrEqual;
-                            case BinaryOperatorKind.UInt:
-                            case BinaryOperatorKind.ULong:
-                                return BinaryOperationKind.UnsignedGreaterThanOrEqual;
-                            case BinaryOperatorKind.Float:
-                            case BinaryOperatorKind.Double:
-                                return BinaryOperationKind.FloatingGreaterThanOrEqual;
-                            case BinaryOperatorKind.Decimal:
-                                return BinaryOperationKind.DecimalGreaterThanOrEqual;
-                            case BinaryOperatorKind.Pointer:
-                                return BinaryOperationKind.PointerGreaterThanOrEqual;
-                            case BinaryOperatorKind.Enum:
-                                return BinaryOperationKind.EnumGreaterThanOrEqual;
-                            case BinaryOperatorKind.UserDefined:
-                                return BinaryOperationKind.OperatorMethodGreaterThanOrEqual;
-                        }
-
-                        break;
-
-                    case BinaryOperatorKind.GreaterThan:
-                        switch (operatorKind & BinaryOperatorKind.TypeMask)
-                        {
-                            case BinaryOperatorKind.Int:
-                            case BinaryOperatorKind.Long:
-                                return BinaryOperationKind.IntegerGreaterThan;
-                            case BinaryOperatorKind.UInt:
-                            case BinaryOperatorKind.ULong:
-                                return BinaryOperationKind.UnsignedGreaterThan;
-                            case BinaryOperatorKind.Float:
-                            case BinaryOperatorKind.Double:
-                                return BinaryOperationKind.FloatingGreaterThan;
-                            case BinaryOperatorKind.Decimal:
-                                return BinaryOperationKind.DecimalGreaterThan;
-                            case BinaryOperatorKind.Pointer:
-                                return BinaryOperationKind.PointerGreaterThan;
-                            case BinaryOperatorKind.Enum:
-                                return BinaryOperationKind.EnumGreaterThan;
-                            case BinaryOperatorKind.UserDefined:
-                                return BinaryOperationKind.OperatorMethodGreaterThan;
-                        }
-
-                        break;
+                    case CSharp.BinaryOperatorKind.Addition:
+                        return BinaryOperatorKind.Add;
```

Comment:
>return BinaryOperatorKind.Add; [](start = 24, length = 30)

How do we represent string concatenation? Should we ever return Concatenate kind.

Reply:
We return Concatenate kind only for the VB Concatenate operator. I can open a separate issue to track whether or not we should represent C# + operation with string operators as Concatenate.

Reply:
Filed https://github.com/dotnet/roslyn/issues/21737